isis/tests/FunctionalTestsJigsaw.cpp

```diff
@@ -0,0 +1,106 @@
+#include "Fixtures.h"
+#include "Pvl.h"
+#include "PvlGroup.h"
+#include "TestUtilities.h"
+#include "ControlNet.h"
+#include "Statistics.h"
+
+#include "jigsaw.h"
+
+#include "gtest/gtest.h"
+
+using namespace Isis;
+
+static QString APP_XML = FileName("$ISISROOT/bin/xml/jigsaw.xml").expanded();
+
+TEST_F(ObservationPair, FunctionalTestJigsawCamSolveAll) {
+  // delete to remove old camera for when cam is updated
+  delete cubeL;
+  delete cubeR;      
+  
+  QTemporaryDir prefix;
+  QString outCnetFileName = prefix.path() + "/outTemp.net";
+  QVector<QString> args = {"fromlist="+cubeListFile, "cnet="+cnetPath, "onet="+outCnetFileName, 
+                           "observations=yes", "update=yes", "Cksolvedegree=3", 
+                           "Camsolve=all", "twist=no", "Spsolve=none", "Radius=no", "imagescsv=on", "file_prefix="+prefix.path()};
+
+  UserInterface options(APP_XML, args);
+  
+  Pvl log; 
+  try {
+    jigsaw(options, &log);
+  }
+  catch (IException &e) {
+    FAIL() << "Unable to bundle: " << e.what() << std::endl;
+  }
+  
+  // images were updated 
+  cubeL = new Cube(cubeLPath, "r");
+  cubeR = new Cube(cubeRPath, "r");
+  
+  ControlNet oNet;
+  oNet.ReadControl(outCnetFileName);
+
+  EXPECT_NEAR(oNet.AverageResidual(), 0.123132, 0.00001); 
+  EXPECT_NEAR(oNet.GetMaximumResidual(), 0.379967, 0.00001);
+  ASSERT_EQ(oNet.GetNumIgnoredMeasures(), 0);
+  ASSERT_EQ(oNet.GetNumValidPoints(), 46);
+  
+  QList<ControlPoint*> points = oNet.GetPoints(); 
+  
+  QVector<double> x(points.size()); 
+  QVector<double> y(points.size()); 
+  QVector<double> z(points.size()); 
+  
+  for (int i = 0; i < points.size(); i++) { 
+      x.append(points.at(i)->GetAdjustedSurfacePoint().GetX().kilometers());
+      y.append(points.at(i)->GetAdjustedSurfacePoint().GetY().kilometers());
+      z.append(points.at(i)->GetAdjustedSurfacePoint().GetZ().kilometers());
+  }
+  
+  Statistics xstats; 
+  xstats.AddData(x.data(), x.size());
+  EXPECT_NEAR(xstats.Average(),           778.32403157284591, 0.00001);
+  EXPECT_NEAR(xstats.StandardDeviation(), 782.62477485560999, 0.00001);
+  EXPECT_NEAR(xstats.Minimum(),           0,                  0.00001);
```

Comment:
Seems somewhat concerning that mins are all 0?

Reply:
That is concerning. You should check and see which points are 0 in the output network. It could be that these are the fixed points in the network.